File: bubblesort.py
def bubblesort(lst):
    #from last index to second
    for passes in range(len(lst) - 1, 0, -1):
        #from [0,passes[ keep swapping to put the largest
        #number at index passes
        for i in range(passes):
            if lst[i] > lst[i+1]:
                swap(lst, i, i+1)
    return lst
            
def swap(lst, i, j):
    temp = lst[i]
    lst[i] = lst[j]
    lst[j] = temp
    
print "{0}".format(bubblesort([23,57,75,33,6,8,56]))

File: bubblesort.java
import java.util.Arrays;
import java.util.List;

public class BubbleSort {

	public static List<Integer> bubbleSort(List<Integer> lst){
		//go from last to second number
		for(int passes= lst.size() -1; passes != 1; --passes){
			//go from [0, passes[
			for(int i=0; i != passes; ++i){
				if(lst.get(i) > lst.get(i+1)){
					swap(lst, i, i+1);
				}
			}
		}
		return lst;
	}//end bubblesort
	
	public static void swap(List<Integer> lst, int i, int j){
		int temp = lst.get(i);
		lst.set(i, lst.get(j));
		lst.set(j, temp);
	}//end swap
	
	public static void main(String[] args) {
		List<Integer> list = Arrays.asList(23,57,75,33,6,8,56);
		System.out.println( bubbleSort(list).toString() );
	}
}
